compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -3828,6 +3828,30 @@ private STNode parseExpressionRhsInternal(OperatorPrecedence currentPrecedenceLe
                     isInConditionalExpr);
         }
 
+        if (isSimpleTypeNode(lhsExpr.kind)) {
```

Comment:
I think parsing is correct in this scenario. `byte[]` is same as `Person[]`. Both are index-access expressions.
i.e:
```ballerina
type Person record {|
|}

public function main() {
    var x = Person[]; // ----(1)
    var y = byte[]; // ----(2)
}
```
Both (1) and (2) are identical, in terms of syntax and semantics.

If we consider either of (1) 0r (2), compiler should give two errors:
- Parser would give an error for missing index.
- Semantic analyzer would give an error saying `invalid operation: type 'typedesc<int>' does not support indexing`

Reply:
I think this needs to be fixed at semantic analyzer